omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ReportDesignResource.java

```
@@ -0,0 +1,111 @@
+import org.openmrs.module.reporting.report.service.ReportService;
+import org.openmrs.module.reportingrest.web.controller.ReportingRestController;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import java.util.List;
+
+/**
+ * {@link Resource} for {@link ReportDesign}s, supporting standard CRUD operations
+ */
+@Resource(name = RestConstants.VERSION_1 + ReportingRestController.REPORTING_REST_NAMESPACE + "/designs",
+supportedClass = ReportDesign.class, supportedOpenmrsVersions = {"1.8.* - 9.9.*"})
+public class ReportDesignResource extends DelegatingCrudResource<ReportDesign> {
+
+  @Override
+  public ReportDesign newDelegate() {
+    return new ReportDesign();
+  }
+
+  @Override
+  public ReportDesign getByUniqueId(String uuid) {
+    return getReportService().getReportDesignByUuid(uuid);
+  }
+
+  @Override
+  protected PageableResult doSearch(RequestContext context) {
+    String reportDefinitionUuid = context.getParameter("reportDefinitionUuid");
+    if (StringUtils.isBlank(reportDefinitionUuid)) {
+      throw new IllegalArgumentException("reportDefinitionUuid parameter is required");
+    }
+
+    ReportDefinition reportDefinition = getReportDefinitionService().getDefinitionByUuid(reportDefinitionUuid);
```

What happens when no report definition is found with the given uuid?

---

I've changed it to return an empty result, as "no report definition found, then there are no report requests related to that not existing report definition in the system", previously it would return all requests.
But I guess, an exception/404 would be an option too.

---

@mseaton are there any OpenMRS suitable classes that we can use for return when parameter not found? Instead throwing exception like throw new IllegalArgumentException("reportDefinitionUuid parameter is required"); ?

---

I'm not sure @druchniewicz , but I think throwing an exception makes sense.  Assuming things are coded right on the front-end, you'll never run into this, as the UI should always be running reports based on the list of available reports that exist, right?  So this isn't really something we need a strong user-facing response around, is it?

---

@mseaton exactly, so I'll leave it as it is with exception thrown

---

Resolved

---

What I meant here was that `if (reportDefinition == null)` then we should throw an exception, not return an empty result, as the only way this would happen would be if the user supplied an invalid reportDefinition, which is different than supplying a valid reportDefinition that just doesn't have any designs configured.

---

I've corrected it